src/test/java/se/liu/ida/hefquin/engine/data/impl/VocabularyMappingTest.java

```diff
@@ -0,0 +1,113 @@
+package se.liu.ida.hefquin.engine.data.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.vocabulary.OWL;
+import org.apache.jena.vocabulary.RDF;
+import org.junit.Test;
+
+import se.liu.ida.hefquin.engine.data.Triple;
+import se.liu.ida.hefquin.engine.data.VocabularyMapping;
+import se.liu.ida.hefquin.engine.query.SPARQLGraphPattern;
+import se.liu.ida.hefquin.engine.query.SPARQLUnionPattern;
+import se.liu.ida.hefquin.engine.query.TriplePattern;
+import se.liu.ida.hefquin.engine.query.impl.SPARQLUnionPatternImpl;
+import se.liu.ida.hefquin.engine.query.impl.TriplePatternImpl;
+
+public class VocabularyMappingTest {
+
+	@Test
+	public void VocabularyMappingConstructorTest() {
+		final Set<Triple> testTriples = CreateTestTriples();
+		final List<Triple> triplesList = new ArrayList<Triple>();
+		triplesList.addAll(testTriples);
+		final VocabularyMapping vm = new VocabularyMappingImpl(testTriples);
+		
+		final Node s = NodeFactory.createVariable("s");
+		final Node p = NodeFactory.createVariable("p");
+		final Node o = NodeFactory.createVariable("o");
+		final org.apache.jena.graph.Triple testQuery = new TriplePatternImpl(s, p, o).asJenaTriple();
+		List<Triple> QueryResults = new ArrayList<Triple>();
+		final Iterator<org.apache.jena.graph.Triple> i = vm.getVocabularyMappingAsGraph().find(testQuery);
+		while(i.hasNext()) {
+			final Triple t = new TripleImpl(i.next());
+			QueryResults.add(t);
+		}
+		//assertEquals(QueryResults, triplesList);
```

Comment:
Why did this not work? Did the test fail? You may need to implement the `equals` method in `Triple` for this `assertEquals` to work correctly.

Reply:
If I'm guessing, this comparison did not work because I was creating new triples from the query results. I changed this to sets of Jena Triples instead so that the results could be compared directly

Reply:
Just to expand a bit on the `assertEquals(Object expected, Object actual)` method. The only think it does is to check if both are not `null` and then call `expected.equals(actual)`. In the case of lists, the equals method is implemented in such a way that it iterates through both lists and checks element by element if they are equal. This is why I first suggested to implement the `equals` method in `Triple`. Since Jena Triples already have an implementation for `equals` it seems that now it works :)